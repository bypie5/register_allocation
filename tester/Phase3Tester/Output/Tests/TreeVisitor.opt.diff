--- SelfTestCases/TreeVisitor.opt.out	2011-04-28 14:55:20.000000000 -0700
+++ ./Output/Tests/TreeVisitor.opt.out	2020-02-24 15:38:14.000000000 -0800
@@ -1,43 +1 @@
-16
-100000000
-4
-8
-12
-14
-16
-20
-24
-28
-100000000
-50000000
-333
-333
-333
-28
-24
-333
-20
-16
-333
-333
-333
-14
-12
-8
-333
-4
-100000000
-1
-1
-1
-0
-1
-4
-8
-14
-16
-20
-24
-28
-0
-0
+Program has no "Main" function.
