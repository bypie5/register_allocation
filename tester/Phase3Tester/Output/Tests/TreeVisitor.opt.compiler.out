func Main [in 0, out 0, local 0]
$s0 = 
PrintIntS($s0)
ret

func TV.Start [in 0, out 0, local 0]
$s0 = HeapAllocZ(24)
if $s0 goto :null1
Error()
null1:
$s0 = 
if $s1 goto :null2
Error()
null2:
$s0 = 
PrintIntS(100000000)
if $s1 goto :null3
Error()
null3:
$s0 = 
if $s1 goto :null4
Error()
null4:
$s0 = 
if $s1 goto :null5
Error()
null5:
$s0 = 
if $s1 goto :null6
Error()
null6:
$s0 = 
if $s1 goto :null7
Error()
null7:
$s0 = 
if $s1 goto :null8
Error()
null8:
$s0 = 
if $s1 goto :null9
Error()
null9:
$s0 = 
if $s1 goto :null10
Error()
null10:
$s0 = 
PrintIntS(100000000)
$s0 = HeapAllocZ(12)
$s2 = $s0
PrintIntS(50000000)
if $s1 goto :null11
Error()
null11:
$s0 = 
PrintIntS(100000000)
if $s2 goto :null12
Error()
null12:
$s0 = 
PrintIntS($s0)
if $s1 goto :null13
Error()
null13:
$s0 = 
PrintIntS($s0)
if $s1 goto :null14
Error()
null14:
$s0 = 
PrintIntS($s0)
if $s1 goto :null15
Error()
null15:
$s0 = 
PrintIntS($s0)
if $s1 goto :null16
Error()
null16:
$s0 = 
PrintIntS($s0)
if $s1 goto :null17
Error()
null17:
$s0 = 
if $s1 goto :null18
Error()
null18:
$s0 = 
if $s1 goto :null19
Error()
null19:
$s1 = 
PrintIntS($s1)
ret

func Tree.Init [in 0, out 0, local 0]
ret

func Tree.SetRight [in 0, out 0, local 0]
ret

func Tree.SetLeft [in 0, out 0, local 0]
ret

func Tree.GetRight [in 0, out 0, local 0]
ret

func Tree.GetLeft [in 0, out 0, local 0]
ret

func Tree.GetKey [in 0, out 0, local 0]
ret

func Tree.SetKey [in 0, out 0, local 0]
ret

func Tree.GetHas_Right [in 0, out 0, local 0]
ret

func Tree.GetHas_Left [in 0, out 0, local 0]
ret

func Tree.SetHas_Left [in 0, out 0, local 0]
ret

func Tree.SetHas_Right [in 0, out 0, local 0]
ret

func Tree.Compare [in 0, out 0, local 0]
$s0 = 0
$s0 = Add($s2 1)
$s0 = LtS($s1 $s2)
if $s0 goto :if1_else
$s0 = 0
goto :if1_end
if1_else:
$s2 = LtS($s1 $s0)
if $s2 goto :if2_else
$s0 = 0
goto :if2_end
if2_else:
$s0 = 1
if2_end:
if1_end:
ret

func Tree.Insert [in 0, out 0, local 0]
$s0 = HeapAllocZ(24)
if $s0 goto :null20
Error()
null20:
$s2 = 
$s0 = $s3
$s0 = 1
while1_top:
if $s0 goto :while1_end
if $s0 goto :null21
Error()
null21:
$s0 = 
$s3 = LtS($s1 $s0)
if $s3 goto :if3_else
if $s2 goto :null22
Error()
null22:
$s0 = 
if $s0 goto :if4_else
if $s1 goto :null23
Error()
null23:
$s0 = 
goto :if4_end
if4_else:
$s0 = 0
if $s1 goto :null24
Error()
null24:
$s0 = 
if $s1 goto :null25
Error()
null25:
$s1 = 
if4_end:
goto :if3_end
if3_else:
if $s0 goto :null26
Error()
null26:
$s0 = 
if $s0 goto :if5_else
if $s1 goto :null27
Error()
null27:
$s0 = 
goto :if5_end
if5_else:
$s0 = 0
if $s1 goto :null28
Error()
null28:
$s0 = 
if $s1 goto :null29
Error()
null29:
$s1 = 
if5_end:
if3_end:
goto :while1_top
while1_end:
ret

func Tree.Delete [in 0, out 0, local 0]
$s0 = $s1
$s0 = $s1
$s0 = 1
$s0 = 0
$s0 = 1
while2_top:
if $s0 goto :while2_end
if $s1 goto :null30
Error()
null30:
$s0 = 
$s3 = LtS($s1 $s0)
if $s3 goto :if6_else
if $s2 goto :null31
Error()
null31:
$s0 = 
if $s0 goto :if7_else
$s0 = $s1
if $s1 goto :null32
Error()
null32:
$s0 = 
goto :if7_end
if7_else:
$s0 = 0
if7_end:
goto :if6_end
if6_else:
$s2 = LtS($s0 $s1)
if $s2 goto :if8_else
if $s3 goto :null33
Error()
null33:
$s0 = 
if $s0 goto :if9_else
$s0 = $s1
if $s1 goto :null34
Error()
null34:
$s0 = 
goto :if9_end
if9_else:
$s0 = 0
if9_end:
goto :if8_end
if8_else:
if $s0 goto :if10_else
if $s1 goto :null35
Error()
null35:
$s0 = 
if $s0 goto :if11_else
if $s1 goto :null36
Error()
null36:
$s1 = 
if $s1 goto :if11_else
$s0 = 1
goto :if11_end
if11_else:
$s2 = 
if11_end:
goto :if10_end
if10_else:
$s2 = 
if10_end:
$s0 = 1
$s0 = 0
if8_end:
if6_end:
$s0 = 0
goto :while2_top
while2_end:
ret

func Tree.Remove [in 0, out 0, local 0]
if $s0 goto :null37
Error()
null37:
$s0 = 
if $s0 goto :if12_else
$s0 = 
goto :if12_end
if12_else:
if $s0 goto :null38
Error()
null38:
$s0 = 
if $s0 goto :if13_else
$s0 = 
goto :if13_end
if13_else:
if $s0 goto :null39
Error()
null39:
$s0 = 
if $s2 goto :null40
Error()
null40:
$s1 = 
if $s1 goto :null41
Error()
null41:
$s3 = 
$s0 = 
if $s0 goto :if14_else
if $s4 goto :null42
Error()
null42:
$s0 = 
if $s2 goto :null43
Error()
null43:
$s0 = 
goto :if14_end
if14_else:
if $s0 goto :null44
Error()
null44:
$s0 = 
if $s2 goto :null45
Error()
null45:
$s0 = 
if14_end:
if13_end:
if12_end:
ret

func Tree.RemoveRight [in 0, out 0, local 0]
while3_top:
if $s0 goto :null46
Error()
null46:
$s0 = 
if $s0 goto :while3_end
if $s1 goto :null47
Error()
null47:
if $s0 goto :null48
Error()
null48:
$s1 = 
if $s1 goto :null49
Error()
null49:
$s1 = 
$s0 = 
$s0 = $s2
if $s2 goto :null50
Error()
null50:
$s0 = 
goto :while3_top
while3_end:
if $s0 goto :null51
Error()
null51:
$s0 = 
if $s2 goto :null52
Error()
null52:
$s0 = 
ret

func Tree.RemoveLeft [in 0, out 0, local 0]
while4_top:
if $s0 goto :null53
Error()
null53:
$s0 = 
if $s0 goto :while4_end
if $s1 goto :null54
Error()
null54:
if $s0 goto :null55
Error()
null55:
$s1 = 
if $s1 goto :null56
Error()
null56:
$s1 = 
$s0 = 
$s0 = $s2
if $s2 goto :null57
Error()
null57:
$s0 = 
goto :while4_top
while4_end:
if $s0 goto :null58
Error()
null58:
$s0 = 
if $s2 goto :null59
Error()
null59:
$s0 = 
ret

func Tree.Search [in 0, out 0, local 0]
$s0 = $s1
$s0 = 1
$s0 = 0
while5_top:
if $s0 goto :while5_end
if $s1 goto :null60
Error()
null60:
$s0 = 
$s3 = LtS($s1 $s0)
if $s3 goto :if15_else
if $s2 goto :null61
Error()
null61:
$s0 = 
if $s0 goto :if16_else
if $s1 goto :null62
Error()
null62:
$s0 = 
goto :if16_end
if16_else:
$s0 = 0
if16_end:
goto :if15_end
if15_else:
$s2 = LtS($s0 $s1)
if $s2 goto :if17_else
if $s3 goto :null63
Error()
null63:
$s0 = 
if $s0 goto :if18_else
if $s1 goto :null64
Error()
null64:
$s0 = 
goto :if18_end
if18_else:
$s0 = 0
if18_end:
goto :if17_end
if17_else:
$s0 = 1
$s0 = 0
if17_end:
if15_end:
goto :while5_top
while5_end:
ret

func Tree.Print [in 0, out 0, local 0]
$s0 = $s1
$s2 = 
ret

func Tree.RecPrint [in 0, out 0, local 0]
if $s0 goto :null65
Error()
null65:
$s0 = 
if $s0 goto :if19_else
if $s1 goto :null66
Error()
null66:
$s1 = 
$s0 = 
goto :if19_end
if19_else:
$s0 = 1
if19_end:
if $s0 goto :null67
Error()
null67:
$s0 = 
PrintIntS($s0)
if $s1 goto :null68
Error()
null68:
$s0 = 
if $s0 goto :if20_else
if $s1 goto :null69
Error()
null69:
$s1 = 
$s0 = 
goto :if20_end
if20_else:
$s0 = 1
if20_end:
ret

func Tree.accept [in 0, out 0, local 0]
PrintIntS(333)
if $s0 goto :null70
Error()
null70:
$s0 = 
ret

func Visitor.visit [in 0, out 0, local 0]
if $s0 goto :null71
Error()
null71:
$s0 = 
if $s0 goto :if21_else
if $s1 goto :null72
Error()
null72:
$s1 = 
if $s0 goto :null73
Error()
null73:
$s0 = 
goto :if21_end
if21_else:
$s0 = 0
if21_end:
if $s0 goto :null74
Error()
null74:
$s0 = 
if $s0 goto :if22_else
if $s1 goto :null75
Error()
null75:
$s1 = 
if $s0 goto :null76
Error()
null76:
$s0 = 
goto :if22_end
if22_else:
$s0 = 0
if22_end:
ret

func MyVisitor.visit [in 0, out 0, local 0]
if $s0 goto :null77
Error()
null77:
$s0 = 
if $s0 goto :if23_else
if $s1 goto :null78
Error()
null78:
$s1 = 
if $s0 goto :null79
Error()
null79:
$s0 = 
goto :if23_end
if23_else:
$s0 = 0
if23_end:
if $s0 goto :null80
Error()
null80:
$s0 = 
PrintIntS($s0)
if $s1 goto :null81
Error()
null81:
$s0 = 
if $s0 goto :if24_else
if $s1 goto :null82
Error()
null82:
$s1 = 
if $s0 goto :null83
Error()
null83:
$s0 = 
goto :if24_end
if24_else:
$s0 = 0
if24_end:
ret

